src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,241 @@
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
+    )
+
+
+@impl
+def tox_add_core_config(core_conf, state):
+    opt = state.conf.options
+
+    if opt.current_env or opt.print_deps_to or opt.print_extras_to:
+        # We do not want to install the main package.
+        # no_package is the same as skipsdist.
+        loader = MemoryLoader(no_package=True)
+        core_conf.loaders.insert(0, loader)
+
+    if opt.current_env:
+        opt.default_runner = "current-env"
+        return
+
+    if getattr(opt.print_deps_to, "name", object()) == getattr(
+        opt.print_extras_to, "name", object()
+    ):
+        raise RuntimeError(
+            "The paths given to --print-deps-to and --print-extras-to cannot be identical."
+        )
+
+    if opt.print_deps_to or opt.print_extras_to:
+        opt.default_runner = "print-env"
+        return
+
+    # No options used - switch back to the standard runner
+    # Workaround for: https://github.com/tox-dev/tox/issues/2264
+    opt.default_runner = "virtualenv"
+
+
+@impl
+def tox_add_env_config(env_conf, state):
+    opt = state.conf.options
+    # This allows all external commands.
+    # All of them are external for us.
+    # Because tox 4 no longer reads $TOX_TESTENV_PASSENV,
+    # this plugin always passes all environment variables by default.
+    if opt.current_env:
+        allow_external_cmds = MemoryLoader(allowlist_externals=["*"], pass_env=["*"])
+        env_conf.loaders.insert(0, allow_external_cmds)
+    # For print-deps-to and print-extras-to, use empty
+    # list of commands so the tox does nothing.
+    if opt.print_deps_to or opt.print_extras_to:
+        empty_commands = MemoryLoader(commands=[], commands_pre=[], commands_post=[])
+        env_conf.loaders.insert(0, empty_commands)
+
+
+class Installer:
+    """Noop installer"""
+
+    def install(self, *args, **kwargs):
+        return None
+
+
+class CurrentEnvLocalSubProcessExecutor(Execute):
+    def build_instance(
+        self,
+        request,
+        options,
+        out,
+        err,
+    ):
+        request.env["PATH"] = ":".join(
```

Comment:
This seems like a bad approach to me, why not set this via set_env instead? This works but is ugly and not used as intended 🤔 Also does not work on Windows?

Reply:
FWIW We don't support Windows. If Windows user supply PRs we gladly review them but that's it.